modules/misunderstood/src/hooks/after_incoming_middleware/00_misunderstood_flag.js

```diff
@@ -1,27 +1,51 @@
 const axios = require('axios')
 const _ = require('lodash')
 
-const flag = async () => {
-  const matchesQuickReply = async () => {
-    const { currentFlow } = event.state.context
-    if (currentFlow && currentFlow.startsWith('skills/choice')) {
-      const flow = await bp.ghost.forBot(event.botId).readFileAsObject('flows', currentFlow)
-      const preview = event.preview.toLowerCase()
-      const kw = _.chain(flow.skillData.keywords)
-        .values()
-        .flatten()
-        .find(kw => kw.toLowerCase() === preview.toLowerCase())
-        .value()
-
-      if (kw) {
-        return true
-      }
+const { currentFlow, currentNode } = event.state.context
+
+const matchesQuickReply = async (flow) => {
+  if (currentFlow && currentFlow.startsWith('skills/choice')) {
+    const preview = event.preview.toLowerCase()
+    const kw = _.chain(flow.skillData.keywords)
+      .values()
+      .flatten()
+      .find(kw => kw.toLowerCase() === preview.toLowerCase())
+      .value()
+
+    if (kw) {
+      return true
+    }
+  }
+
+  return false
+}
+
+const inWaitNode = async (flow) => {
+  if (currentFlow && currentNode) {
+    const node = _.find(flow.nodes, node => node.name === currentNode)
+    
+    if (node && node.onReceive != null) {
+      return true
     }
+  }
+
+  return false
+}
+
+const isMisunderstood = async () => {
+  if (event.type === 'text' && event.nlu.intent.name === 'none'){
+    const flow = currentFlow && await bp.ghost.forBot(event.botId).readFileAsObject('flows', currentFlow)
 
-    return false
+    if (await matchesQuickReply(flow) || await inWaitNode(flow)) {
+      return false;
+    }
```

Comment:
There's a small cleanup possible here. You pass the flow as an argument, but you don't even use it in your methods..
```js
if (!flow || await matchesQuickReply(flow) || await inWaitNode(flow)) {
  return false;
}
```

 Then you can expect `flow` to be always defined in your other methods, so
```js
const inWaitNode = async (flow) => {
  // your currentNode check is unnecessary, because the find will not find it if it's undefined (but I think you'll always have a node)
  const node =  flow.nodes.find(x => x.name === currentNode)
  return node && node.onReceive != null
}

and 
```js
const matchesQuickReply = async (flow) => {
  if (flow.startsWith('skills/choice')) {
   /// ...
  }
}
```